packages/provider/src/transports/wallet-request-handler.ts

```diff
@@ -41,7 +41,12 @@ export interface WalletSignInOptions {
 }
 
 export class WalletRequestHandler implements ExternalProvider, JsonRpcHandler, ProviderMessageRequestHandler {
-  private signer: Signer | null
+  // signer interface of the wallet. A null value means there is no signer (ie. user not signed in). An undefined
+  // value means the signer state is unknown, usually meaning the wallet app is booting up and initializing. Of course
+  // a Signer value is the actually interface to a signed-in account
+  private signer: Signer | null | undefined
```

Comment:
Alternatively, I think this could be a `Promise<Signer> | undefined` instead.

Reply:
Never mind, it seems like it's something more complicated than a basic promise...